org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/MessageTracer.java

```diff
@@ -11,31 +11,34 @@
  ******************************************************************************/
 package org.eclipse.lsp4j.jsonrpc;
 
-import org.eclipse.lsp4j.jsonrpc.TracingMessageConsumer.RequestMetadata;
-
 import java.io.PrintWriter;
 import java.time.Clock;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
 import java.util.function.Function;
 
+import org.eclipse.lsp4j.jsonrpc.TracingMessageConsumer.RequestMetadata;
+import org.eclipse.lsp4j.jsonrpc.json.MessageJsonHandler;
+
 /**
  * Wraps a {@link MessageConsumer} with one that logs in a way that the LSP Inspector can parse. *
  * https://microsoft.github.io/language-server-protocol/inspector/
  */
 public class MessageTracer implements Function<MessageConsumer, MessageConsumer> {
 	private final PrintWriter printWriter;
+	private final MessageJsonHandler jsonHandler;
 	private final Map<String, RequestMetadata> sentRequests = new HashMap<>();
 	private final Map<String, RequestMetadata> receivedRequests = new HashMap<>();
 
-	MessageTracer(PrintWriter printWriter) {
+	MessageTracer(PrintWriter printWriter, MessageJsonHandler jsonHandler) {
```

Comment:
This is also a breaking API change.

Reply:
The constructor is package private, so I would say this is not a breaking API change

Reply:
My bad. You are right, of course. But I'd still prefer a setter to the constructor in this case.